frame/support/procedural/src/lib.rs

```diff
@@ -987,7 +966,18 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
 /// Items that lack a `syn::Ident` for whatever reason are first checked to see if they exist,
 /// verbatim, in the local/destination trait before they are copied over, so you should not need to
 /// worry about collisions between identical unnamed items.
-#[import_tokens_attr(frame_support::macro_magic)]
+#[import_tokens_attr {
+    format!(
+        "{}::macro_magic",
+        match generate_crate_access_2018("frame-support") {
+            Ok(path) => Ok(path),
+            Err(_) => generate_crate_access_2018("frame"),
+        }
+        .expect("Failed to find either `frame-support` or `frame` in `Cargo.toml` dependencies.")
+        .to_token_stream()
+        .to_string()
+    )
+}]
```

Comment:
what part is `ugly` to you? the panic or the big attribute size?

maybe you can define it outside the attribute as a function which returns a `String` so it is more readable.

also it would be better to combine with the error message returned by `generate_crate_access_2018` if possible as they are very precise.

How is the error message currently? this is the most important part for me.

But I think if we support a `Result<String, proc_macro::TokenStream>` it would allow to write more idiomatic code, and better error message.
(Or even `Result<proc_macro::TokenStream, proc_macro::TokenStream>`  maybe)